tests/golem/resource/test_resourcehandshake.py

```diff
@@ -101,19 +106,22 @@ def test_request_task_handshake(self, *_):
     def test_request_task_success(self, *_):
         self.session._handshake_required = Mock()
         self.session.send = Mock()
+        wtct_dict = {k: v for k, v in self.message.items()}
+        wtct_dict.update({
+            'task_header': self.task_header.to_dict()
+        })
 
         self.session._handshake_required.return_value = False
         self.session.request_task(**self.message)
 
-        assert not self.session.disconnect.called
-        assert self.session.send.called
-
-        slots = self.session.send.call_args[0][0].__slots__
+        self.assertFalse(self.session.disconnect.called)
+        self.assertTrue(self.session.send.called)
 
-        msg = message.tasks.WantToComputeTask(**self.message)
-        msg_slots = msg.__slots__
+        sent_wtct = self.session.send.call_args[0][0]
 
-        assert slots == msg_slots
+        msg = message.tasks.WantToComputeTask(**wtct_dict)
+        self.assertEqual(sent_wtct.__slots__, msg.__slots__)
```

Comment:
I know it's just a rewrite, but this never made sense. `__slots__` doesn't return slot values, just `__slots__` attribute. 
```suggestion
        self.assertEqual(sent_wtct, msg)
```